src/SFA.DAS.ApprenticeCommitments.Jobs.Functions/ApprenticeshipChangedEventHandler.cs

```diff
@@ -0,0 +1,42 @@
+﻿using NServiceBus;
+using SFA.DAS.ApprenticeCommitments.Jobs.Api;
+using SFA.DAS.ApprenticeCommitments.Jobs.Functions.Infrastructure;
+using SFA.DAS.ApprenticeCommitments.Messages.Events;
+using SFA.DAS.Notifications.Messages.Commands;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.ApprenticeCommitments.Jobs.Functions
+{
+    public class ApprenticeshipChangedEventHandler : IHandleMessages<ApprenticeshipChangedEvent>
+    {
+        private readonly ApplicationSettings settings;
+        private readonly IEcsApi api;
+
+        public ApprenticeshipChangedEventHandler(ApplicationSettings settings, IEcsApi api)
+        {
+            this.api = api;
+            this.settings = settings;
+        }
+
+        public async Task Handle(ApprenticeshipChangedEvent message, IMessageHandlerContext context)
+        {
+            if (!settings.Notifications.Templates.TryGetValue("ApprenticeshipChangedEmail", out var templateId))
+                throw new System.Exception("Missing configuration `Notifications:Templates:ApprenticeshipChangedEmail`");
+
+            var url = $"{settings.ApprenticeCommitmentsWeb.BaseUrl}/Apprenticeships";
+
+            var apprentice = await api.GetApprentice(message.ApprenticeId);
+
+            await context.Send(new SendEmailCommand(
+                templateId,
+                apprentice.Email.ToString(),
+                new Dictionary<string, string>
+                {
+                    { "GivenName", apprentice.FirstName },
+                    { "FamilyName", apprentice.LastName },
+                    { "ConfirmApprenticeshipUrl", url },
```

Comment:
It worries me that, we are only publishing the event 'ApprenticeshipChangedEvent' when the last seen event is more than 24 hours. I think that logic really belongs here.

Sorry I missed this in the API layer. 

It's just this is a public event and the logic to determine if the email should be sent or not really does belong here

Reply:
Yes, I agree.  I've moved that login into this project now.